src/qt/bitcoingui.cpp

```diff
@@ -1105,6 +1105,15 @@ void BitcoinGUI::detectShutdown()
     }
 }
 
+void BitcoinGUI::shouldQuit()
```

Comment:
Please don't make the qt shutdown sequence even more complicated :(

Wouldn't a better idea be to add StartShutdown here to app.requestShutdown, called from https://github.com/bitcoin/bitcoin/blob/master/src/qt/bitcoin.cpp#L691